# IS 362 – Week 5 Assignment

In [1]:
import pandas as pd

In [2]:
airports = pd.read_csv('https://raw.githubusercontent.com/tidyverse/nycflights13/refs/heads/main/data-raw/airports.csv')
weather = pd.read_csv('https://raw.githubusercontent.com/tidyverse/nycflights13/refs/heads/main/data-raw/weather.csv')

The link above will extract the CSV data from the files that were uploaded to GitHub, however, the data itself must be raw in order to properly generate a Pandas DataFrame. The extracted data will be given names as variables.

The tests below are done to determine if the data was generated properly, showing both the first five and last five rows of data.

In [3]:
airports.head()

faa                           name        lat        lon   alt  tz dst  \
0  04G              Lansdowne Airport  41.130472 -80.619583  1044  -5   A   
1  06A  Moton Field Municipal Airport  32.460572 -85.680028   264  -6   A   
2  06C            Schaumburg Regional  41.989341 -88.101243   801  -6   A   
3  06N                Randall Airport  41.431912 -74.391561   523  -5   A   
4  09J          Jekyll Island Airport  31.074472 -81.427778    11  -5   A   

              tzone  
0  America/New_York  
1   America/Chicago  
2   America/Chicago  
3  America/New_York  
4  America/New_York

In [4]:
airports.tail()

faa                       name        lat         lon   alt  tz dst  \
1453  ZUN                 Black Rock  35.083228 -108.791778  6454  -7   A   
1454  ZVE     New Haven Rail Station  41.298669  -72.925992     7  -5   A   
1455  ZWI  Wilmington Amtrak Station  39.736667  -75.551667     0  -5   A   
1456  ZWU   Washington Union Station  38.897460  -77.006430    76  -5   A   
1457  ZYP               Penn Station  40.750500  -73.993500    35  -5   A   

                 tzone  
1453    America/Denver  
1454  America/New_York  
1455  America/New_York  
1456  America/New_York  
1457  America/New_York

## Question 1: What is the northernmost airport in the United States?

To be able to get the desired data, I would have to be able to generate a list of the first five northernmost airports to compare them to one another (using '.head()'), and then choose the airport on the top row. In general, the northernmost value would be the highest latitude ('lat') value in its series. So, I had to sort the DataFrame by the latitude series with descending values, so that they would be ordered from the highest value to the lowest value. By sorting the DataFrame this way, as well as having it display the first five rows, I would read the top row and see the airport name with the highest latitude value, indicating that it's the northernmost airport in the United States.

It should be noted that for latitude, positive values would indicate that the location is north of the equator in the northern hemisphere, while a negative values would indicate the southern hemosphire, south of the equator. Note, that the equator would be given the value of 0.

In [5]:
airports.sort_values('lat', ascending=False).head()

faa                                        name        lat         lon  \
417  EEN                     Dillant Hopkins Airport  72.270833   42.898333   
230  BRW                  Wiley Post Will Rogers Mem  71.285446 -156.766003   
110  AIN                          Wainwright Airport  70.638056 -159.994722   
708  K03                               Wainwright As  70.613378 -159.860350   
152  ATK  Atqasuk Edward Burnell Sr Memorial Airport  70.467300 -157.436000   

     alt  tz dst              tzone  
417  149  -5   A                NaN  
230   44  -9   A  America/Anchorage  
110   41  -9   A  America/Anchorage  
708   35  -9   A  America/Anchorage  
152   96  -9   A  America/Anchorage

### Answer:

Based on the information in the DataFrame, the northernmost airport in the United States is **Dillant Hopkins Airport** in Swanzey, New Hampshire. It has a latitude of 72.270833°N, as shown in the DataFrame in the 'lat' series, with an index number of 417. Having the highest latitude value, it's considered the northernmost airport listed in the current DataFrame.

## Question 2: What is the easternmost airport in the United States?

While it is simpler trying to find the northernmost airport in the United States, finding the easternmost airport might be a bit more complicated. I cannot use the same method for when finding the northernmost aiport, as longitudes can be either a positive or a negative number when marking coordinates east (positive) or west (negative) from the prime meridian. And since the values can go up to either 360 or -360, it can represent either a eastern or western coordinate. Though, usually longitiude coordinates go up to 180 or -180 to avoid complications. 

I tested it for the highest and lowest values for the 'lon' series in the DataFrame, with airports listed in the top row associated with those highest/lowest values being located in Alaska, west of the mainland United States. 

While Alaska is considered both the easternmost and westernmost part of the country, according to some sources and the coordinates used. But basing this on the world map, it's actually ***westernmost***.

The easternmost airport would be located on the easternmost part of the country, which is Maine. Meaning the easternmost airport in the DataFrame would be located in Maine, but to locate it, I would have to consider another factor regarding geography: **timezones**.

To workaround this issue, I would have to take into account the correlation of the coordinate system and the timezones. First, I had to create a variable to filter the 'airports' DataFrame based on timezones less than 0, which is reprsented through the series 'tz.' This would reorder them starting from 0 and counting through negative numbers, the easternmost part of the United States has -5 as its timezone value.

Then I would sort the variable 'easternmost' based on the longitude/'lon' series, while showing the 5 lowest values for longitude as reordered as the first 5 five rows in the DataFrame. Thus, it would give me the desired information in the first row.

In [6]:
easternmost = airports[airports.tz <= 0]
easternmost

faa                           name        lat         lon   alt  tz dst  \
0     04G              Lansdowne Airport  41.130472  -80.619583  1044  -5   A   
1     06A  Moton Field Municipal Airport  32.460572  -85.680028   264  -6   A   
2     06C            Schaumburg Regional  41.989341  -88.101243   801  -6   A   
3     06N                Randall Airport  41.431912  -74.391561   523  -5   A   
4     09J          Jekyll Island Airport  31.074472  -81.427778    11  -5   A   
...   ...                            ...        ...         ...   ...  ..  ..   
1453  ZUN                     Black Rock  35.083228 -108.791778  6454  -7   A   
1454  ZVE         New Haven Rail Station  41.298669  -72.925992     7  -5   A   
1455  ZWI      Wilmington Amtrak Station  39.736667  -75.551667     0  -5   A   
1456  ZWU       Washington Union Station  38.897460  -77.006430    76  -5   A   
1457  ZYP                   Penn Station  40.750500  -73.993500    35  -5   A   

                 tzone  
0     America/New_York  
1      America/Chicago  
2      America/Chicago  
3     America/New_York  
4     America/New_York  
...                ...  
1453    America/Denver  
1454  America/New_York  
1455  America/New_York  
1456  America/New_York  
1457  America/New_York  

[1456 rows x 8 columns]

In [7]:
easternmost.sort_values('lon').tail()

faa                        name        lat         lon  alt  tz dst  \
259   CAR                Caribou Muni  46.871500  -68.017917  626  -5   A   
624   HUL                Houlton Intl  46.123083  -67.792056  489  -5   A   
444   EPM  Eastport Municipal Airport  44.910111  -67.012694   45  -5   A   
417   EEN     Dillant Hopkins Airport  72.270833   42.898333  149  -5   A   
1290  SYA                Eareckson As  52.712275  174.113620   98  -9   A   

                  tzone  
259    America/New_York  
624    America/New_York  
444    America/New_York  
417                 NaN  
1290  America/Anchorage

### Answer:

Based on the information in the DataFrame generated, the easternmost airport in the United States is **Caribou Municipal Airport** in Caribou, Maine. This is because of the correlation of the timezones and longitude, given that the airport's timezone is -5 and has a longitude of 68.017917°W (-68.017917), indicating that it is the easternmost airport in the generated DataFrame. Plus, to reiterate, the easternmost part of the United States is the state of Maine, so with the airport located in the state, it can be concluded that this is the correct answer.

Overall, the correlation between the longitude ('lat') and timezone ('tz') values helped me filter the data and sort it to receive the correct answer.

## Question 3: On February 12th, 2013, which New York area airport had the windiest weather?

Much like determining the easternmost airport of the United States, I would have to apply the same principle when finding which New York area airport had the windiest weather. First, I would have to filter the DataFrame based on New York area airports, being LaGuardia Airport (LGA) and John F. Kennedy Airport (JFK). First, I had to confirm their presence in the CSV file, so I filtered both LGA and JFK in the DataFrame. Sure enough, they were there. Each DataFrame generated having 8706 rows for both airports, indicating that there are 8706 counts of recorded weather data for each airport, for almost every hour each day of the year.

In [8]:
weather[weather.origin == 'LGA']

origin  year  month  day  hour   temp   dewp  humid  wind_dir  \
17409    LGA  2013      1    1     1  39.92  26.06  57.33     260.0   
17410    LGA  2013      1    1     2  41.00  26.06  54.97     260.0   
17411    LGA  2013      1    1     3  41.00  26.06  54.97     260.0   
17412    LGA  2013      1    1     4  41.00  26.06  54.97     260.0   
17413    LGA  2013      1    1     5  39.92  24.98  54.81     250.0   
...      ...   ...    ...  ...   ...    ...    ...    ...       ...   
26110    LGA  2013     12   30    14  35.96  19.94  51.78     340.0   
26111    LGA  2013     12   30    15  33.98  17.06  49.51     330.0   
26112    LGA  2013     12   30    16  32.00  15.08  49.19     340.0   
26113    LGA  2013     12   30    17  30.92  12.92  46.74     320.0   
26114    LGA  2013     12   30    18  28.94  10.94  46.41     330.0   

       wind_speed  wind_gust  precip  pressure  visib             time_hour  
17409    13.80936   23.01560     0.0    1011.9   10.0  2013-01-01T06:00:00Z  
17410    17.26170   25.31716     0.0    1011.5   10.0  2013-01-01T07:00:00Z  
17411    16.11092   24.16638     0.0    1012.0   10.0  2013-01-01T08:00:00Z  
17412    17.26170   25.31716     0.0    1011.8   10.0  2013-01-01T09:00:00Z  
17413    14.96014   21.86482     0.0    1011.4   10.0  2013-01-01T10:00:00Z  
...           ...        ...     ...       ...    ...                   ...  
26110    13.80936   21.86482     0.0    1017.1   10.0  2013-12-30T19:00:00Z  
26111    17.26170   21.86482     0.0    1018.8   10.0  2013-12-30T20:00:00Z  
26112    14.96014   23.01560     0.0    1019.5   10.0  2013-12-30T21:00:00Z  
26113    17.26170        NaN     0.0    1019.9   10.0  2013-12-30T22:00:00Z  
26114    18.41248        NaN     0.0    1020.9   10.0  2013-12-30T23:00:00Z  

[8706 rows x 15 columns]

In [9]:
weather[weather.origin == 'JFK']

origin  year  month  day  hour   temp   dewp  humid  wind_dir  \
8703     JFK  2013      1    1     1  39.02  26.06  59.37     260.0   
8704     JFK  2013      1    1     2  39.02  26.06  59.37     270.0   
8705     JFK  2013      1    1     3  39.92  26.96  59.50     260.0   
8706     JFK  2013      1    1     4  39.92  28.04  62.21     250.0   
8707     JFK  2013      1    1     5  39.02  26.96  61.63     260.0   
...      ...   ...    ...  ...   ...    ...    ...    ...       ...   
17404    JFK  2013     12   30    14  37.94  21.02  50.14     330.0   
17405    JFK  2013     12   30    15  35.96  17.96  47.57     340.0   
17406    JFK  2013     12   30    16  33.98  15.98  47.25     330.0   
17407    JFK  2013     12   30    17  32.00  12.92  44.74     320.0   
17408    JFK  2013     12   30    18  30.02  10.04  42.66     340.0   

       wind_speed  wind_gust  precip  pressure  visib             time_hour  
8703     12.65858        NaN     0.0    1012.6   10.0  2013-01-01T06:00:00Z  
8704     11.50780        NaN     0.0    1012.4   10.0  2013-01-01T07:00:00Z  
8705     14.96014        NaN     0.0    1012.7   10.0  2013-01-01T08:00:00Z  
8706     17.26170        NaN     0.0    1012.6   10.0  2013-01-01T09:00:00Z  
8707     14.96014        NaN     0.0    1012.1   10.0  2013-01-01T10:00:00Z  
...           ...        ...     ...       ...    ...                   ...  
17404    16.11092   24.16638     0.0    1017.4   10.0  2013-12-30T19:00:00Z  
17405    17.26170   25.31716     0.0    1018.7   10.0  2013-12-30T20:00:00Z  
17406    19.56326   27.61872     0.0    1019.5   10.0  2013-12-30T21:00:00Z  
17407    13.80936        NaN     0.0    1020.1   10.0  2013-12-30T22:00:00Z  
17408    18.41248        NaN     0.0    1020.9   10.0  2013-12-30T23:00:00Z  

[8706 rows x 15 columns]

After confirming the presence of LGA and JFK, a new DataFrame must be generated, filtered to have both LGA and JFK present in it. This new DataFrame will be used to sort the wind speed values.

In [10]:
wind = weather[weather.origin.isin(['LGA', 'JFK'])]
wind

origin  year  month  day  hour   temp   dewp  humid  wind_dir  \
8703     JFK  2013      1    1     1  39.02  26.06  59.37     260.0   
8704     JFK  2013      1    1     2  39.02  26.06  59.37     270.0   
8705     JFK  2013      1    1     3  39.92  26.96  59.50     260.0   
8706     JFK  2013      1    1     4  39.92  28.04  62.21     250.0   
8707     JFK  2013      1    1     5  39.02  26.96  61.63     260.0   
...      ...   ...    ...  ...   ...    ...    ...    ...       ...   
26110    LGA  2013     12   30    14  35.96  19.94  51.78     340.0   
26111    LGA  2013     12   30    15  33.98  17.06  49.51     330.0   
26112    LGA  2013     12   30    16  32.00  15.08  49.19     340.0   
26113    LGA  2013     12   30    17  30.92  12.92  46.74     320.0   
26114    LGA  2013     12   30    18  28.94  10.94  46.41     330.0   

       wind_speed  wind_gust  precip  pressure  visib             time_hour  
8703     12.65858        NaN     0.0    1012.6   10.0  2013-01-01T06:00:00Z  
8704     11.50780        NaN     0.0    1012.4   10.0  2013-01-01T07:00:00Z  
8705     14.96014        NaN     0.0    1012.7   10.0  2013-01-01T08:00:00Z  
8706     17.26170        NaN     0.0    1012.6   10.0  2013-01-01T09:00:00Z  
8707     14.96014        NaN     0.0    1012.1   10.0  2013-01-01T10:00:00Z  
...           ...        ...     ...       ...    ...                   ...  
26110    13.80936   21.86482     0.0    1017.1   10.0  2013-12-30T19:00:00Z  
26111    17.26170   21.86482     0.0    1018.8   10.0  2013-12-30T20:00:00Z  
26112    14.96014   23.01560     0.0    1019.5   10.0  2013-12-30T21:00:00Z  
26113    17.26170        NaN     0.0    1019.9   10.0  2013-12-30T22:00:00Z  
26114    18.41248        NaN     0.0    1020.9   10.0  2013-12-30T23:00:00Z  

[17412 rows x 15 columns]

After filtering the data for LGA and JFK, the DataFrame would have to be filtered again in order to show the weather data for February 12, 2023, for both airports.

In [11]:
date = wind[(wind.year == 2013) & (wind.month == 2) & (wind.day == 12)]
date

origin  year  month  day  hour   temp   dewp  humid  wind_dir  \
9709     JFK  2013      2   12     0  42.08  37.04  82.17     280.0   
9710     JFK  2013      2   12     1  39.02  35.96  88.66     250.0   
9711     JFK  2013      2   12     2  41.00  28.94  61.89     280.0   
9712     JFK  2013      2   12     3  41.00  26.06  54.97     290.0   
9713     JFK  2013      2   12     4  39.92  24.98  54.81     280.0   
9714     JFK  2013      2   12     5  39.02  26.96  61.63     260.0   
9715     JFK  2013      2   12     6  39.92  24.98  54.81     280.0   
9716     JFK  2013      2   12     7  39.92  26.06  57.33     270.0   
9717     JFK  2013      2   12     8  42.08  24.08  48.56     280.0   
9718     JFK  2013      2   12     9  42.08  24.98  50.42     270.0   
9719     JFK  2013      2   12    10  42.98  24.08  46.91     280.0   
9720     JFK  2013      2   12    11  44.06  24.98  46.73     280.0   
9721     JFK  2013      2   12    12  44.06  24.98  46.73     280.0   
9722     JFK  2013      2   12    13  44.96  24.08  43.49     280.0   
9723     JFK  2013      2   12    14  46.94  24.08  40.35     310.0   
9724     JFK  2013      2   12    15  44.96  23.00  41.56     290.0   
9725     JFK  2013      2   12    16  42.08  23.00  46.41     270.0   
9726     JFK  2013      2   12    17  41.00  24.08  50.63     260.0   
9727     JFK  2013      2   12    18  41.00  24.08  50.63     270.0   
9728     JFK  2013      2   12    19  39.92  23.00  50.46     280.0   
9729     JFK  2013      2   12    20  39.02  21.92  49.93     290.0   
9730     JFK  2013      2   12    21  37.94  21.92  52.09     300.0   
9731     JFK  2013      2   12    22  37.94  21.92  52.09     290.0   
9732     JFK  2013      2   12    23  37.04  21.92  53.97     280.0   
18415    LGA  2013      2   12     0  42.08  37.40  86.89       NaN   
18416    LGA  2013      2   12     1  44.06  30.92  59.64     270.0   
18417    LGA  2013      2   12     2  42.98  26.06  50.94     290.0   
18418    LGA  2013      2   12     3  42.08  26.06  52.73     290.0   
18419    LGA  2013      2   12     4  41.00  24.98  52.56     270.0   
18420    LGA  2013      2   12     5  41.00  24.98  52.56     290.0   
18421    LGA  2013      2   12     6  42.08  24.08  48.56     270.0   
18422    LGA  2013      2   12     7  41.00  24.08  50.63     260.0   
18423    LGA  2013      2   12     8  42.08  23.00  46.41     260.0   
18424    LGA  2013      2   12     9  42.08  23.00  46.41     260.0   
18425    LGA  2013      2   12    10  42.98  23.00  44.83     270.0   
18426    LGA  2013      2   12    11  42.98  23.00  44.83     270.0   
18427    LGA  2013      2   12    12  44.06  23.00  43.02     270.0   
18428    LGA  2013      2   12    13  44.06  23.00  43.02     300.0   
18429    LGA  2013      2   12    14  44.06  23.00  43.02     300.0   
18430    LGA  2013      2   12    15  44.06  21.92  41.10     260.0   
18431    LGA  2013      2   12    16  42.98  24.08  46.91     300.0   
18432    LGA  2013      2   12    17  42.98  23.00  44.83     260.0   
18433    LGA  2013      2   12    18  41.00  21.92  46.23     260.0   
18434    LGA  2013      2   12    19  41.00  21.92  46.23     260.0   
18435    LGA  2013      2   12    20  39.92  23.00  50.46     290.0   
18436    LGA  2013      2   12    21  39.02  21.92  49.93     310.0   
18437    LGA  2013      2   12    22  39.02  23.00  52.26     310.0   
18438    LGA  2013      2   12    23  37.94  23.00  54.51     300.0   

       wind_speed  wind_gust  precip  pressure  visib             time_hour  
9709      4.60312        NaN     0.0    1007.1   10.0  2013-02-12T05:00:00Z  
9710     10.35702        NaN     0.0    1007.3    7.0  2013-02-12T06:00:00Z  
9711     17.26170   24.16638     0.0    1007.7   10.0  2013-02-12T07:00:00Z  
9712     16.11092   26.46794     0.0    1008.5   10.0  2013-02-12T08:00:00Z  
9713     12.65858   21.86482     0.0    1009.0   10.0  2013-02-12T09:00:00Z  
9714     12.65858        NaN     0.0    1009.2   10.0  201

After generating this DataFrame to filter in information for February 12, 2013, another one will be generated to sort the wind speed values in a descending manner, with the first row showing the highest wind speed value. The information in that row will also show which New York area airport had the windiest weather that day.

In [12]:
date.sort_values('wind_speed', ascending=False)

origin  year  month  day  hour   temp   dewp  humid  wind_dir  \
18417    LGA  2013      2   12     2  42.98  26.06  50.94     290.0   
18428    LGA  2013      2   12    13  44.06  23.00  43.02     300.0   
18429    LGA  2013      2   12    14  44.06  23.00  43.02     300.0   
9717     JFK  2013      2   12     8  42.08  24.08  48.56     280.0   
18427    LGA  2013      2   12    12  44.06  23.00  43.02     270.0   
18416    LGA  2013      2   12     1  44.06  30.92  59.64     270.0   
9716     JFK  2013      2   12     7  39.92  26.06  57.33     270.0   
18433    LGA  2013      2   12    18  41.00  21.92  46.23     260.0   
18425    LGA  2013      2   12    10  42.98  23.00  44.83     270.0   
9721     JFK  2013      2   12    12  44.06  24.98  46.73     280.0   
9719     JFK  2013      2   12    10  42.98  24.08  46.91     280.0   
9718     JFK  2013      2   12     9  42.08  24.98  50.42     270.0   
18430    LGA  2013      2   12    15  44.06  21.92  41.10     260.0   
18420    LGA  2013      2   12     5  41.00  24.98  52.56     290.0   
9711     JFK  2013      2   12     2  41.00  28.94  61.89     280.0   
18418    LGA  2013      2   12     3  42.08  26.06  52.73     290.0   
9715     JFK  2013      2   12     6  39.92  24.98  54.81     280.0   
9723     JFK  2013      2   12    14  46.94  24.08  40.35     310.0   
9724     JFK  2013      2   12    15  44.96  23.00  41.56     290.0   
9712     JFK  2013      2   12     3  41.00  26.06  54.97     290.0   
9725     JFK  2013      2   12    16  42.08  23.00  46.41     270.0   
18432    LGA  2013      2   12    17  42.98  23.00  44.83     260.0   
9727     JFK  2013      2   12    18  41.00  24.08  50.63     270.0   
9726     JFK  2013      2   12    17  41.00  24.08  50.63     260.0   
9722     JFK  2013      2   12    13  44.96  24.08  43.49     280.0   
9720     JFK  2013      2   12    11  44.06  24.98  46.73     280.0   
18434    LGA  2013      2   12    19  41.00  21.92  46.23     260.0   
18423    LGA  2013      2   12     8  42.08  23.00  46.41     260.0   
18426    LGA  2013      2   12    11  42.98  23.00  44.83     270.0   
18424    LGA  2013      2   12     9  42.08  23.00  46.41     260.0   
18419    LGA  2013      2   12     4  41.00  24.98  52.56     270.0   
18421    LGA  2013      2   12     6  42.08  24.08  48.56     270.0   
18422    LGA  2013      2   12     7  41.00  24.08  50.63     260.0   
9730     JFK  2013      2   12    21  37.94  21.92  52.09     300.0   
9729     JFK  2013      2   12    20  39.02  21.92  49.93     290.0   
9714     JFK  2013      2   12     5  39.02  26.96  61.63     260.0   
9713     JFK  2013      2   12     4  39.92  24.98  54.81     280.0   
18437    LGA  2013      2   12    22  39.02  23.00  52.26     310.0   
9728     JFK  2013      2   12    19  39.92  23.00  50.46     280.0   
9710     JFK  2013      2   12     1  39.02  35.96  88.66     250.0   
18435    LGA  2013      2   12    20  39.92  23.00  50.46     290.0   
18436    LGA  2013      2   12    21  39.02  21.92  49.93     310.0   
18438    LGA  2013      2   12    23  37.94  23.00  54.51     300.0   
18431    LGA  2013      2   12    16  42.98  24.08  46.91     300.0   
9731     JFK  2013      2   12    22  37.94  21.92  52.09     290.0   
9732     JFK  2013      2   12    23  37.04  21.92  53.97     280.0   
9709     JFK  2013      2   12     0  42.08  37.04  82.17     280.0   
18415    LGA  2013      2   12     0  42.08  37.40  86.89       NaN   

       wind_speed  wind_gust  precip  pressure  visib             time_hour  
18417    23.01560   31.07106     0.0    1007.1   10.0  2013-02-12T07:00:00Z  
18428    21.86482   25.31716     0.0    1011.7   10.0  2013-02-12T18:00:00Z  
18429    20.71404   25.31716     0.0    1011.5   10.0  2013-02-12T19:00:00Z  
9717     20.71404   27.61872     0.0    1011.6   10.0  2013-02-12T13:00:00Z  
18427    19.56326   26.46794     0.0    1011.8   10.0  2013-02-12T17:00:00Z  
18416    19.56326   28.76950     0.0    1006.6   10.0  201

### Answer:

 **LaGuardia Airport** (**LGA**) is the New York area airport to have had the windiest weather on February 12, 2023 during  7 AM. Its wind speed value was 23.01560 mph, with a wind gust value of 31.07106 mph.

Considering the temperature utilizes the imperial system measurement, Fahrenheit, it can be assumed that the wind speed also utilizes imperial units as well (mph; **m**iles **p**er **h**our).